Tasks/UseDotNetV2/models.ts

```diff
@@ -100,16 +103,55 @@ export class VersionParts {
         }
     }
 
-    private static ValidateVersionSpec(version): boolean {
+    /**
+     * Validate the version if this string is a explicit version number. Returns an exception if the version number is not explicit.
+     * @param version the input version number as string
+     */
+    private static ValidateExplicitVersionNumber(version: string): void {
         try {
             let parts = version.split('.');
             // validate version
-            if (parts.length < 2 || parts.length > 3 || (parts[1] == "x" && parts.length > 2) || (parts[1] != "x" && parts.length <= 2) || !parts[0] || !parts[1] || (parts.length == 3 && !parts[2]) || Number.isNaN(Number.parseInt(parts[0])) || (Number.isNaN(Number.parseInt(parts[1])) && parts[1] != "x")) {
-                throw "";
+            if ((parts.length != 3) || // check if the version has 3 parts                
+                !parts[0] || // The major version must always be set
+                !parts[1] || // The minor version must always be set
+                !parts[2] || // The patch version must always be set                
+                Number.isNaN(Number.parseInt(parts[0])) || // the major version number must be a number
+                Number.isNaN(Number.parseInt(parts[1])) || // the minor version number must be a number
+                Number.isNaN(Number.parseInt(parts[2].split('-')[0])) // the patch version number must be a number. (the patch version can have a '-' because of version numbers like: 1.0.0-beta-50)                                   
+
+            ) {
+                throw tl.loc("VersionNumberHasTheWrongFormat", version);
             }
+            new semver.Range(version);
+        }
+        catch (ex) {
+            throw tl.loc("VersionNotAllowed", version)
```

Comment:
Why do we need this catch statement? We can just let the original error flow here given we dont  expect any error other than VersionNumberHasTheWrongFormat

Reply:
because the line `new semver.Range(version);` can also throw an error.

Reply:
Got it. Do we miss anything by hiding the original failure. Should it also get logged.

Reply:
Now I logged the inner error :)